tamr_unify_client/mastering/published_cluster/configuration.py

```diff
@@ -42,3 +44,46 @@ def __repr__(self):
             f"relative_id={self.relative_id!r}, "
             f"versions_time_to_live={self.versions_time_to_live!r})"
         )
+
+
+class PublishedClustersConfigurationBuilder:
+    """A builder object to modify an existing published cluster configuration.
+
+    :param cluster_configuration: The cluster configuration to modify.
+    :type cluster_configuration: :class:`~tamr_unify_client.mastering.published_cluster.configuration.PublishedClustersConfiguration`
+    """
+
+    def __init__(self, cluster_configuration):
+        self._data = copy.deepcopy(cluster_configuration._data)
```

Comment:
This design is fine for mutable builder, but we should refactor for immutable builders.